diagnostic.md

```diff
@@ -105,18 +157,53 @@ JSON.
 ```
 
 ```sh
-# your answer here
+curl --request PATCH 7 http://www.example.com/donuts/42 \
+--header "Content-Type: application/json" \
+--data '{
+    "name": "Krüller"
+    "done": true
+}'
 ```
 
 Write an AJAX request to change the donut on the server using JSON.
 
 ```js
-let changeDonut = /* your answer here */;
+let changeDonut = function(event) {
+  event.preventDefault();
+  $.ajax({
+    url: 'http://www.example.com/donuts/42',
+    type: 'PATCH',
+    data: {
+      "name": "Krüller"
+      'done': true,
+      'list_id': 42
+    }
+  })
+  .done(logResponseBody)
+  .fail(logRequestError);
+};;
 ```
 
 Now, rewrite your AJAX request to use FormData. Assume you have already targeted
 the correct HTML form before registering your AJAX function with the form.
 
 ```js
-let changeDonut = /* your answer here */;
+let changeDonut = function(event) {
+  event.preventDefault();
+  let itemId = $itemIdInput.val();
```

Comment:
what is itemIdInput here, it is a reference to a jQuery object or are you attempting to use jQuery.  If so be sure to wrap your selector in ('selector').